java-client/src/main/java/org/opensearch/client/json/JsonpSerializable.java

```diff
@@ -33,11 +33,34 @@
 package org.opensearch.client.json;
 
 import jakarta.json.stream.JsonGenerator;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.io.UncheckedIOException;
 
 /**
  * An object that is its own JsonP serializer
  */
 public interface JsonpSerializable {
 
     void serialize(JsonGenerator generator, JsonpMapper mapper);
+
+    /**
+     * A default method which returns string representation for the instances of classes
+     * implementing JsonpSerializable interface.<br>
+     * Usage : Eg for SearchRequest.class<pre>{@code SearchRequest implements JsonpSerializable{..}
+     * SearchRequest searchRequest = SearchRequest.of(request -> request...);
+     * String searchRequestString = searchRequest.writeValueAsString();}</pre>  <br>
+     *
+     */
+    default String writeValueAsString() {
```

Comment:
I think we have to find better name (`toJson`? `asJson`?) to this method and extract it to a different interface (fe `PlainJsonSerializable`), it does not belong to `JsonpSerializable` anyhow (it uses separate Jsonp provider):

```
public interface PlainJsonSerializable extends JsonpSerializable {
    default String toJson() {
         ...
    }
}
```

Yes, it uses opt-in model (we would need to update classes where we need the plain JSON serialization) but it clearly separates 2 different serialization mechanisms.

Reply:
* For name I think `toJson` should do fine right ? short and defines purpose .
*  For design, my initial idea was to make a default method to handle serialization for all the classes implementing JsonpSerializable , without having to implement any other interface or calling a Utils class, clients can just call searchResponse.toJson() and done , which uses default mapper and abstracts away the procedure of having to call serialize by there own generator and mapper .

Reply:
Okay nvm toJson results in conflict coz JsonData which implements JsonpSerializable has a method implemented with same name

Reply:
> Also serialize can still be used with mapper and generator, our method is just a default implementation of a more generalize method.

That's the thing, the same API (interface) bundles 2 different serialization implementation, and there is no way to optout from it.

Reply:
* I see the point of no way of opting out and separation of concerns  but is it not ideal in this case, do you see any disadvantages in future for this approach.
* Also  for Opt In using separate interface, do we need to re assess the classes we want to implement the new interface or all current classes which are implementing JsonpSerializable.
* Related to the previous point , if new classes are introduced implementing JsonpSerializable we will have to asses, if we want to OptIn right ?

Reply:
> I see the point of no way of opting out and separation of concerns but is it not ideal in this case, do you see any disadvantages in future for this approach.

I don't see any disadvantages to be fair

> Also for Opt In using separate interface, do we need to re assess the classes we want to implement the new interface or all current classes which are implementing JsonpSerializable.

Correct, this is one time job but I truly believe it is worth the effort to keep APIs clean and concise. 

> Related to the previous point , if new classes are introduced implementing JsonpSerializable we will have to asses, if we want to OptIn(Implement new Interface) right ?

The code is supposed to be generated: as far as generator is aware of this change, it won't be  noticeable.